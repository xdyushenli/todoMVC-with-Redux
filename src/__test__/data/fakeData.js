const activeItem = {
    isActive: true,
    id: 1,
    text: 'active item'
}

const completedItem = {
    isActive: false,
    id: 1,
    text: 'completed item'
}

const todoList = [
    {
        isActive: false,
        id: 1,
        text: 'completed item'
    },
    {
        isActive: true,
        id: 2,
        text: 'active item'
    }
]

export {activeItem, completedItem, todoList}