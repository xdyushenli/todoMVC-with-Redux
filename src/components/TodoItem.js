import React from 'react';
import '../index.css';

export default class TodoItem extends React.Component{
    constructor(props) {
        super(props);

        this.state = {
            editing: false,
            inputText: '',
        }
    }

    handleSelectSingle(e) {
        this.props.selectSingle(this.props.item.id, e);
    }

    showInput(e) {
        this.setState({editing: true});
    }

    handleChangeText(e) {
        this.setState({inputText: e.target.value});
    }

    handleInputCompleted(e) {
        let inputText = this.state.inputText;

        if(e.key === 'Enter' || e.type === 'blur') {
            this.setState({
                editing: false,
                inputText: '',
            });
            this.props.changeText(this.props.item.id, inputText, e);
        }
    }

    handleDeleteItem(e) {
        this.props.deleteItem(this.props.item.id, e);
    }

    render() {
        let item = this.props.item;
        let editing = this.state.editing;

        return (
            <li className='todo-item'>
                <button className={item.isActive ? 'finish-btn' : 'finish-btn finished-btn'}
                        onClick={(e) => this.handleSelectSingle(e)}
                ></button>
                <div    className={item.isActive ? 'item-text' : 'item-text finished-text'}
                        onDoubleClick={(e) => this.showInput(e)}
                >{  
                    editing ? 
                    <input  className='change-text' 
                            placeholder={item.text} 
                            onChange={(e) => this.handleChangeText(e)}
                            onKeyPress={(e) => this.handleInputCompleted(e)}
                            onBlur={(e) => this.handleInputCompleted(e)}
                            value={this.state.inputText} /> :
                    item.text
                }</div>
                <button className='delete-btn'
                        onClick={(e) => this.handleDeleteItem(e)}
                >âœ•</button>
            </li>
        );
    }
}