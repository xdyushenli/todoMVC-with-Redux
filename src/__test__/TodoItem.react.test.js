import React from 'react';
import TodoItem from '../components/TodoItem';
import renderer from 'react-test-renderer';
import Enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {activeItem, completedItem} from './data/fakeData';

Enzyme.configure({adapter: new Adapter()});

it('render active item', () => {
    const tree = renderer.create(
        <TodoItem 
            item={activeItem}
        ></TodoItem>
    ).toJSON();
    expect(tree).toMatchSnapshot();
})

it('render completed item', () => {
    const tree = renderer.create(
        <TodoItem 
            item={completedItem}
        ></TodoItem>
    ).toJSON();
    
    expect(tree).toMatchSnapshot();
})

it('simulate edit active item, state should be {editing: true, inputText: "some text"}', () => {
    const component = shallow(
        <TodoItem 
            item={activeItem}
        ></TodoItem>
    );
    component.find('.item-text').simulate('doubleClick');
    component.find('input').simulate('change', {target: {value: 'some text'}})
    
    expect(component.state()).toEqual({editing: true, inputText: 'some text'});
})

it('simulate edit completed item, state should be {editing: true, inputText: "some text"}', () => {
    const component = shallow(
        <TodoItem 
            item={completedItem}
        ></TodoItem>
    );
    component.find('.item-text').simulate('doubleClick');
    component.find('input').simulate('change', {target: {value: 'some text'}})
    
    expect(component.state()).toEqual({editing: true, inputText: 'some text'});
})