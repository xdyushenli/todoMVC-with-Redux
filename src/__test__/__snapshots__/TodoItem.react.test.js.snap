// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render active item 1`] = `
<li
  className="todo-item"
>
  <button
    className="finish-btn"
    onClick={[Function]}
  />
  <div
    className="item-text"
    onDoubleClick={[Function]}
  >
    active item
  </div>
  <button
    className="delete-btn"
    onClick={[Function]}
  >
    ✕
  </button>
</li>
`;

exports[`render completed item 1`] = `
<li
  className="todo-item"
>
  <button
    className="finish-btn finished-btn"
    onClick={[Function]}
  />
  <div
    className="item-text finished-text"
    onDoubleClick={[Function]}
  >
    completed item
  </div>
  <button
    className="delete-btn"
    onClick={[Function]}
  >
    ✕
  </button>
</li>
`;
